src/psyclone/tests/psyir/symbols/datasymbol_test.py

```diff
@@ -250,23 +251,42 @@ def test_datasymbol_constant_value_setter():
 
     '''
 
-    # Test with valid constant value
+    # Test with valid constant values
     sym = DataSymbol('a', DataType.INTEGER, constant_value=7)
-    assert sym.constant_value == 7
+    assert sym.constant_value.datatype == DataType.INTEGER
+    assert sym.constant_value.value == "7"
     sym.constant_value = 9
-    assert sym.constant_value == 9
+    assert sym.constant_value.datatype == DataType.INTEGER
+    assert sym.constant_value.value == "9"
 
     sym = DataSymbol('a', DataType.REAL, constant_value=3.1415)
-    assert sym.constant_value == 3.1415
+    assert sym.constant_value.datatype == DataType.REAL
+    assert sym.constant_value.value == "3.1415"
     sym.constant_value = 1.0
-    assert sym.constant_value == 1.0
+    assert sym.constant_value.datatype == DataType.REAL
+    assert sym.constant_value.value == "1.0"
 
+    # Test with invalid constant values
     sym = DataSymbol('a', DataType.DEFERRED)
     with pytest.raises(ValueError) as error:
         sym.constant_value = 1.0
     assert ("Error setting 'a' constant value. Constant values are not "
             "supported for 'DataType.DEFERRED' datatypes." in str(error.value))
 
+    # Test with valid constant expressions
+    lhs = Literal('2', DataType.INTEGER)
+    rhs = Reference('constval')
+    ct_expr = BinaryOperation.create(BinaryOperation.Operator.ADD, lhs, rhs)
+    sym = DataSymbol('a', DataType.INTEGER, constant_value=ct_expr)
```

Comment:
Please could you check that the newly-created DataSymbol has the expected properties.

Reply:
Also, the docstring for the DataSymbol constructor needs updating now that `constant_value` can be PSyIR.

Reply:
Added asserts and updated the DataSymbol constructor docstring.